test/box/gh-5928-several-listen-sockets.result

```diff
@@ -0,0 +1,429 @@
+test_run:cmd("setopt delimiter ';'")
+ | ---
+ | - true
+ | ...
+for i = 1, 18 do
+    test_run:cmd(string.format("start server test with args=\"1 %s\"", i))
+    connection_result = check_connection_ex()
+    assert(connection_result == true)
+    test_run:eval('test', string.format("box.cfg{ listen = \'%s\' }", default_server_addr))
+    test_run:cmd("stop server test")
+end;
+ | ---
+ | ...
+test_run:cmd("setopt delimiter ''");
+ | ---
+ | - true
+ | ...
+
+test_run:cmd(string.format("start server test with args=\"%s\"", 1))
+ | ---
+ | - true
+ | ...
+test_run:cmd("switch test")
+ | ---
+ | - true
+ | ...
+test_run:cmd("setopt delimiter ';'")
+ | ---
+ | - true
+ | ...
+function listen_with_bad_uri(idx)
+    local default_server_addr = box.cfg.listen
+    local bad_uri_key = {"uri"}
+    local bad_uri = {}
+    bad_uri[bad_uri_key] = default_server_addr
+    local table_uries = {
+        { "  " },
+        { { uri = string.format("%s", default_server_addr) }, { uri = "  " } },
+        { { uri = string.format("%s", default_server_addr) }, { uri = "?" } },
+        { uri = string.format("%s", default_server_addr), transport = "unexpected_value" },
+        { default_server_addr, uri = string.format("%s", default_server_addr), transport = "plain" },
+        default_server_addr .. "?transport=",
+        default_server_addr .. "?transport=plain&plain",
+        default_server_addr .. "?unexpected_option=unexpected_value",
+        default_server_addr .. "?transport=plain,plain",
+        "?/transport=plain",
+        { transport="plain" },
+        { { uri = string.format("%s", default_server_addr) }, { uri = "" } },
+        default_server_addr .. "?transport=plain=transport=plain",
+        default_server_addr .. "?transport=plain?transport=plain",
+        default_server_addr .. ", " .. default_server_addr,
+        default_server_addr .. ", " .. "?",
+        default_server_addr .. ", " .. "?" .. default_server_addr,
+        default_server_addr .. ", " .. "?," .. default_server_addr,
+        ",?/transport=plain",
+        ",&/transport=plain",
+        { uri = { default_server_addr } },
+        { unexpected_option = "unexpected_value", uri = default_server_addr },
+        { uri = default_server_addr .. "?transport=plain", transport = { "plain, plain" } },
+        bad_uri,
+        "/???",
+        default_server_addr .. "?",
+        default_server_addr .. "&",
+        default_server_addr .. " ",
+        default_server_addr .. "=",
+        default_server_addr .. ";",
+        default_server_addr .. ",",
+        default_server_addr .. "transport=plain" .. ", " .. default_server_addr .. "?, " .. default_server_addr .. "A",
+    }
+    assert(idx and idx > 0 and idx <= #table_uries)
+    box.cfg ({ listen = table_uries[idx] })
+end;
+ | ---
+ | ...
+function get_corresponding_error(idx)
+    local table_uries_corresponding_errors = {
+        "Incorrect value for option 'listen': expected host:service or /unix.socket",
```

Comment:
Try to store the error messages and the bad URIs in together. Otherwise it is very hard to see which errors are for which URIs.

Reply:
fixed